src/components/GoalContent.tsx

```diff
@@ -0,0 +1,206 @@
+                                                    kind={deps.kind}
+                                                    goalId={id}
+                                                    isEmpty={deps.goals.length === 0}
+                                                />
+                                            ))}
+                                        </GoalDependencyListByKind>
+                                    )),
+                                )}
+
+                                {nullable(lastStateComment, (value) => (
+                                    <CommentView
+                                        pin
+                                        id={value.id}
+                                        author={value.activity?.user}
+                                        description={value.description}
+                                        state={value.state}
+                                        createdAt={value.createdAt}
+                                        reactions={value.reactions}
+                                        onSubmit={onGoalCommentSubmit(value)}
+                                        onReactionToggle={onGoalCommentReactionToggle(value.id)}
+                                        onDelete={onGoalCommentDelete(value.id)}
+                                    />
+                                ))}
+                            </>
+                        }
+                        footer={
+                            <CommentCreateForm
+                                states={_isEditable ? project?.flow.states : undefined}
+                                stateId={commentDraft?.stateId}
+                                description={commentDraft?.description}
+                                onSubmit={onGoalCommentCreate}
+                                onCancel={onGoalCommentCancel}
+                                onChange={onGoalCommentChange}
+                            />
+                        }
+                        renderCommentItem={(value) => (
+                            <CommentView
+                                id={value.id}
+                                author={value.activity?.user}
+                                description={value.description}
+                                state={value.state}
+                                createdAt={value.createdAt}
+                                highlight={value.id === highlightCommentId}
+                                reactions={value.reactions}
+                                onSubmit={onGoalCommentSubmit(value)}
+                                onReactionToggle={onGoalCommentReactionToggle?.(value.id)}
+                                onDelete={onGoalCommentDelete?.(value.id)}
+                            />
+                        )}
+                    />
+                ))}
+
+                {nullable(goal, (g) =>
+                    nullable(g._isEditable, () => (
```

Comment:
`g?._isEditable`

Reply:
GoalEditForm & GoalDeleteModal don't accept goal prop when its undefined, this check is necessary.

Reply:
goal prop must be required